msgp/read.go

```diff
@@ -759,6 +759,45 @@ func (m *Reader) ReadBytes(scratch []byte) (b []byte, err error) {
 	return
 }
 
+// ReadBytesHeader reads the size header
+// of a MessagePack 'bin' object. The user
+// is responsible for dealing with the next
+// 'sz' bytes from the reader in an application-specific
+// way.
+func (m *Reader) ReadBytesHeader() (sz uint32, err error) {
+	var p []byte
+	p, err = m.r.Peek(1)
+	if err != nil {
+		return
+	}
+	switch p[0] {
+	case mbin8:
+		p, err = m.r.Next(2)
```

Comment:
[`Next()` does not alter the parse state if an error is returned.](https://github.com/philhofer/fwd/blob/master/reader.go#L221)

But if you read the header successfully, then yes, you can't restart back before the header.